Hazel/src/Platform/OpenGL/OpenGLShader.cpp

```diff
@@ -200,6 +200,18 @@ namespace Hazel {
 		glUseProgram(0);
 	}
 
+	GLint OpenGLShader::GetUniformLocation(const std::string& name) const
+	{
+		HZ_PROFILE_FUNCTION();
+
+		if (m_UniformLocationCache.find(name) != m_UniformLocationCache.end())
+			return m_UniformLocationCache[name];
```

Comment:
you're looking up the cache twice.  would it be slightly more efficient to store the result of find, and then if it isn't end() return ->second?

Reply:
Yes, this would work, I'll update the code now.